pkg/virt-controller/watch/drain/evacuation/evacuation_test.go

```diff
@@ -445,6 +446,89 @@ var _ = Describe("Evacuation", func() {
 
 			controller.Execute()
 		})
+
+		It("Should not evict new VMIs when the amount of concurrent running migrations reached its maximum capacity", func() {
+			var maxParallelMigrationsPerCluster uint32 = 5
+			config, _, _ := testutils.NewFakeClusterConfigUsingKVConfig(&v1.KubeVirtConfiguration{
+				MigrationConfiguration: &v1.MigrationConfiguration{
+					ParallelMigrationsPerCluster: &maxParallelMigrationsPerCluster,
+				},
+			})
+
+			controller = evacuation.
+				NewEvacuationController(
+					vmiInformer,
+					migrationInformer,
+					nodeInformer,
+					podInformer,
+					recorder,
+					virtClient,
+					config)
+
+			nodeName := "node01"
+			addNode(newNode(nodeName))
+
+			vmi1 := newVirtualMachineMarkedForEviction("testvm1", nodeName)
+			vmiFeeder.Add(vmi1)
+
+			for i := 1; i <= int(maxParallelMigrationsPerCluster); i++ {
+				migrationFeeder.Add(newMigration(fmt.Sprintf("mig%d", i), vmi1.Name, v1.MigrationRunning))
```

Comment:
nit: Maybe we can create migrations for fake non-existing VMIs? There's something confusing to me with creating multiple migrations for the same vmi object

edit: I see that in the second test you've created a vmi for every migration - why not to do the same here (or get rid of the vmis altogether if possible)

Reply:
Totally makes sense, done